### ISSUE TITLE
[BUG] Ollama: role should be lowercase

### ISSUE THREAD

**Issue by wkr1337** [langchain4j/langchain4j#1461]
**Describe the bug**
After updating Ollama server to 0.2.5, the /api/chat endpoint changed. The role object inside the messages object became case sensitive. LangChain4j sends all capital case roles to Ollama, but Ollama expects lowercase, and it looks like it is ignoring the content of the request and just randomly answering questions

**To Reproduce**
Here is an example request that used to work before the Ollama update:

{ "model": "llama3", "messages": [ { "role": "USER", "content": "Why is the sky blue?" } ], "options": {}, "stream": false }

But when I run the request now I get what looks like a random response:

{ "model": "llama3", "created_at": "2024-07-14T07:41:45.0417467Z", "message": { "role": "assistant", "content": "I'd be happy to help you with your question. Can you please provide more context or clarify what you mean by \"What is the most effective way to learn English\"? Are you looking for tips on how to improve your grammar, vocabulary, listening, speaking, reading, or writing skills? Or are you wondering about language learning apps, courses, or resources that can help you achieve your goals?\n\nHere are some general tips that might be helpful:\n\n1. **Set specific goals**: Identify what you want to achieve in English, whether it's for personal or professional reasons.\n2. **Practice consistently**: Make time each day to practice speaking, listening, reading, and writing in English.\n3. **Focus on grammar and vocabulary**: Understanding the rules of grammar and building your vocabulary are essential for effective communication.\n4. **Immerse yourself in the language**: Listen to podcasts, watch TV shows or movies, read books or articles, and engage with native speakers to improve your listening and speaking skills.\n5. **Use language learning apps and resources**: There are many great apps, websites, and courses that can help you learn English, such as Duolingo, Coursera, or edX.\n\nLet me know if you have any specific questions or areas of concern, and I'll do my best to assist you!" }, "done_reason": "stop", "done": true, "total_duration": 2718834100, "load_duration": 10797100, "prompt_eval_count": 5, "prompt_eval_duration": 95640000, "eval_count": 266, "eval_duration": 2611441000 }

Then when I change the role "USER" to: "user" it works again.

**Expected behavior**
A response from the Ollama server corresponding to the request, it now looks like random responses.

**Please complete the following information:**
- LangChain4j version: 0.32.0
- LLM(s) used: llama3 (All models through Ollama have this issue)
- Java version: 21
- Spring Boot version (if applicable): 

**Additional context**
Open issue on the Ollama github page: https://github.com/ollama/ollama/issues/5687 
Stackoverflow post: https://stackoverflow.com/questions/78743959/langchain4j-and-ollama-chat-does-not-work-because-of-uppercased-role-value

**Comment by kucharzyk** [comment]
I can confirm this issue. Upgrading ollama from `0.2.1` to `0.2.5` have broken my apps.

I am using this quarkus-langchain4j-ollama and langchain4j 0.3.1

**Comment by langchain4j** [comment]
Thanks a lot for reporting this issue!

**Comment by wkr1337** [comment]
> Thanks a lot for reporting this issue!

It looks like on the Ollama side they are going to make a fix for this by doing role.toLowerCase()

There already is an open Pull request: https://github.com/ollama/ollama/pull/5695/files

**Comment by jmorganca** [comment]
Hi all, it would still be best if we update the roles to be lowercase here as well. Thanks!

**Comment by legalizenet** [comment]
I'm running into a same problem as soon as I restarted ollama which upgraded the version and now my app is returning random response with Llama3/LangChain4J/Ollama . 

- What is the quickest hack on LangChain4J (not sure Changing ENUM class to hack to send lowercase role is an option?) or Ollama side ?
- When can we expect the fixed version of LangChain4j or Ollama with this fix ?

**Comment by langchain4j** [comment]
@legalizenet the fastest is probably either downgrade ollama version or use Ollama OpenAI-compatible API with `langchain4j-open-ai` module.
The fix should be pretty trivial, so the `0.33.0-SNAPSHOT` version should include it during this week.
I also plan a release for the next week.

**Comment by langchain4j** [comment]
Should be fixed by https://github.com/langchain4j/langchain4j/pull/1072

**Comment by langchain4j** [comment]
Fixed, please use `0.33.0-SNAPSHOT` in the meanwhile. You will need to add this to your `pom.xml` to be able to use `SNAPSHOT` version:
```xml
    <repositories>
        <repository>
            <id>snapshot-repo</id>
            <url>https://s01.oss.sonatype.org/content/repositories/snapshots</url>
            <snapshots>
                <enabled>true</enabled>
            </snapshots>
        </repository>
    </repositories>
```

**Comment by legalizenet** [comment]
Thanks for the quick response. For now I have downgraded and works fine. Will get the latest stable release once its out.
